js/util/script_detection.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const ideographicBreakingRegExp = new RegExp([
```

Comment:
Would it make sense to break this regex up into a set of shared strings? For example:

```js
const hangulRegExpString = '...'
const halfwidthFormsRegExpString = '...'

const ideographicBreakingRegExp = new RegExp([
    hangulRegExpString,
    halfwidthFormsRegExpString
].join('|'));
```

I'd be happy to do so. Would you mind defining out which character ranges belong to which sets?